tests/search/logserver/logserver.rb

```diff
@@ -23,6 +23,44 @@ def verify_log_content(content, logserver)
     assert_equal(2, matches)
   end
 
+  def add_loglevel_forward(cfg, level, forward)
+    cfg.add("loglevel", ConfigValue.new(level, ConfigValue.new("forward", forward)))
+  end
+
+  def logd_config_override
+    cfg = ConfigOverride.new("cloud.config.log.logd")
+    add_loglevel_forward(cfg, "event", true)
+    add_loglevel_forward(cfg, "debug", true)
+    add_loglevel_forward(cfg, "spam", true)
+    cfg
+  end
+
+  def get_loglines(args = {})
+    log = ""
+    vespa.logserver.get_vespalog(args) { |data|
+      log << data
+      nil
+    }
+    log.split("\n")
+  end
+
+  def compare_lines(expected_lines, actual_lines)
+    actual_lines.size.times do |i|
+      assert_equal(expected_lines[i], actual_lines[i])
+    end
+  end
+
+  def test_full_logarchive
+    deploy_app(SearchApp.new.sd(SEARCH_DATA+"music.sd").config(logd_config_override))
+    start
+    sleep 2
```

Comment:
Consider doing sleep on loop if log archive is not populated with enough content